src/Orleans.Runtime/MembershipService/MembershipOracle.cs

```diff
@@ -892,6 +924,30 @@ private void OnIAmAliveUpdateInTableTimer(object data)
                 }).Ignore();
         }
 
+        private void OnCleanupEntriesTimer(object data)
+        {
+            var dateLimit = DateTime.UtcNow - this.clusterMembershipOptions.DeleteEntriesOlderThan;
+
+            try
+            {
+                this.membershipTableProvider
+                        .DeleteDeadMembershipTableEntries(dateLimit)
+                        .ContinueWith(task =>
+                        {
+                            if (task.IsFaulted)
+                                this.logger.Error(ErrorCode.MembershipCleanDeadEntriesFailure, "CleanupEntries failed", task.Exception);
+                            return true;
+                        }).Ignore();
+            }
+            catch (MissingMethodException)
```

Comment:
Do we need to also catch `NotImplementedException`?

Reply:
I thought that they were caught in the `Continue` but it seems you are right

Reply:
Maybe we should use an async method instead of `.ContinueWith().Ignore()`, so the error handling is flat